src/cnaas_nms/db/settings.py

```diff
@@ -396,6 +414,32 @@ def check_vlan_collisions(devices_dict: Dict[str, dict], mgmt_vlans: Set[int],
                 device_vlan_names[hostname] = {vxlan_data['vlan_name']}
 
 
+def check_group_priority_collisions(settings: Optional[dict] = None):
+    priorities: Dict[int, str] = {}
+    if not settings:
+        settings, _ = get_group_settings()
+    if settings is None:
```

Comment:
Twice the same check? Perhaps call get_group_settings before calling this function, and pass it along.

```python
Python 3.7.11 (default, Jul 27 2021, 07:03:16) 
[Clang 10.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> settings = None
>>> not settings
True
>>> settings is None
True
>>> 
```

Reply:
get_group_settings should only be called if you don't pass your own settings, so I don't want to call it before. The second check is if get_group_settings returns None, so I think they are both required actually, but I'll change so they both use the same syntax at least :)